### ISSUE TITLE
Multiple file import - Order reversed

### ISSUE THREAD

**Issue by marcluer** [pdfarranger/pdfarranger#936]
Describe the bug
----------------

When importing multiple files, they are imported in reverse order. (last pdf comes first)

To Reproduce
------------

1.  Open PDF arranger
2.  Click on 'Import' Icon
3.  Select multiple pdf files. e.g.
  - Bank-statement_202101.pdf
  - Bank-statement_202102.pdf
  - Bank-statement_202103.pdf
5.  The files are now showing up in the preview in the opposite order
  - ...202103.pdf
  - ...202102.pdf
  - ...202101.pdf
6. When "saving as", the same (reverse order) is kept

Workaround 
-----------------

Inside the import dialog, I sort the files in reverse order (descending).
The files will now be imported in the expected order (ascending).

Expected behavior
-----------------

I would expect the import the files in the order, that is shown in the import dialog.

System and Versions
-------------------

-   PDF Arranger 1.10.00, downloaded from github releases
-   Windows 11

**Comment by kbengs** [comment]
Thanks for reporting. I see the same behavior. In Linux it works as expected.

**Comment by kbengs** [comment]
The flatpak install seems to add them in reversed order too. 

In Windows the order seems to be even more complicated. I can't see the exact logic it use. It has somewhat to do with the order the files are selected in the dialog. This was observed on Windows 10.

@marcluer are the files always added in reversed order, no matter how you select the files in the dialog?

**Comment by marcluer** [comment]
![screenshot-1691871030](https://github.com/pdfarranger/pdfarranger/assets/12187388/fff7a40e-6d09-4349-9738-27fd04fcdbad)
The logic, how the "File name" text box in the import window is populated, is quite strange indeed.
But the "File name" text box does not seem to play any role in the import order.

The actual import order is always reversed to the selected sort order. 
In the above screenshot I would get a consistent import of 5-4-3-2-1.

**Comment by dreua** [comment]
Could this be a Bug in Gtk or Windows?

**Comment by kbengs** [comment]
Yeah they are "just" reversed for me too. Maybe I was too tired yesterday..

> Could this be a Bug in Gtk or Windows?

Yes probably. Maybe FileChooserNative? For Windows we could easily just reverse the order. Not sure how the flatpak could be fixed. Flatpak was tested in KDE.

**Comment by dreua** [comment]
If it happens all the time with flatpaks, we can easily detect and reverse that too. Needs some more testing to see if it is always reversed. Do we also have those issues with drag and drop from the file explorer or is it just with the import?

**Comment by marcluer** [comment]
Drag and drop from Windows Explorer to PDF Arranger window seems to be in completely random order. Even doing the exact same way of selecting the files twice, will generate
a different order.

https://github.com/pdfarranger/pdfarranger/assets/12187388/b8b69fee-a0b8-471f-b1fc-1b4c4c32ac93

**Comment by kbengs** [comment]
> If it happens all the time with flatpaks,

I have tested Fedora flatpak + rpm in KDE + Gnome and Ubuntu flatpak + deb in KDE + GNome. For these the flatpak always reverse the import order whereas the rpm/deb does not.

> Do we also have those issues with drag and drop from the file explorer or is it just with the import?

I think we can do nothing about the drag and drop order. (except adding a sort action)

**Comment by JunglePredator** [comment]
> Drag and drop from Windows Explorer to PDF Arranger window seems to be in completely random order. Even doing the exact same way of selecting the files twice, will generate a different order.


This... is actually why I suggested bulk file renaming sequentially in another post only I didn't realise it was a bug. -.-;; 
It only came into play when updating a PDF using files from an outside source & the filename was random already....

**Comment by varsteel-jeremyzaretski** [comment]
According to Mårten Woxberg in [https://stackoverflow.com/questions/6128962/how-do-i-maintain-the-file-order-in-windows-explorer-when-drag-and-dropping-file](this post):

> If you shift-select a bunch of files from 0000 to 0010 and then drag them into an application (say video files and drag them into VLC). Explorer will feed the list not in the order you selected the items but from the item you left-click-n-dragged from.
> 
> Click on 0000 and they will show up in number order. Click on 0003 and they will show up from 0003 to 0010 and then 0001 and 0002
> 
> So the correct way to "drag n drop" is to always drag from the top.

This is consistent with the demonstration video clip. The drag starts from file 5, so the files are imported in the order 5, 1, 2, 3, 4.

**Comment by kbengs** [comment]
@varsteel-jeremyzaretski thanks.
Status of the issues discussed here should be:
1. Reversed order when importing files with open/import in Windows -> Fixed in main
2. Reversed order when importing files with open/import in Linux with flatpak -> Not fixed. Should be possible to fix.
3. The drag and drop order on Windows I see is as in the post above. Changing this is probably out of scope for this project.
